### DISCUSSION TITLE
How to Nested NavLinks defaultOpened if child active

### DISCUSSION THREAD

**Discussion by goodlythink** [mantinedev/mantine#4993] [Q&A]
In nexjs 13 (app directory)
how to idea set defaultOpened in Nested NavLinks if child active

**Comment by rtivital** [comment]
```tsx
<NavLink defaultOpened={true} />
```

**Comment by goodlythink** [comment]
I mean setting 'defaultOpened' to 'true' when the child 'NavLink' is active.

this my code
import React from 'react'
import { usePathname } from 'next/navigation';

import { NavLink } from '@mantine/core';
import Link from 'next/link';

interface LinksGroupProps {
  icon: React.FC<any>;
  label: string;
  link?: string;
  initiallyOpened?: boolean;
  links?: { label: string; link: string }[];
}

export default function NavLinksGroup({ icon: Icon, label, link, initiallyOpened, links }: LinksGroupProps) {
  const currentRoute = usePathname()
  const hasLinks = Array.isArray(links)


  const items = (hasLinks ? links : []).map(link => {
    return (
      <NavLink
        key={link.label}
        label={link.label}
        component={Link}
        href={link.link}
        active={currentRoute === link.link}
      />
    )
  })

  return (
    <>
      {link ?
        <NavLink
          active={currentRoute === link}
          component={Link}
          href={link}
          label={label}
          leftSection={<Icon size="1.4rem" stroke={1.5} />}
        />
        :
        <NavLink
          label={label}
          leftSection={<Icon size="1.4rem" stroke={1.5} />}
          childrenOffset={34}
          defaultOpened={initiallyOpened}
        >
          {items}
        </NavLink>
      }
    </>
  )
}

**Reply by rtivital** [reply]
You need to determine whether one of the child links is active and set this value in `defaultOpened`. Mantine docs has similar logic, you can find the code in the docs folder.
